diff -r b6d3e557dd16 Eigen/src/Core/arch/CUDA/PacketMathHalf.h
--- a/Eigen/src/Core/arch/CUDA/PacketMathHalf.h	Fri Oct 13 15:58:52 2017 -0700
+++ b/Eigen/src/Core/arch/CUDA/PacketMathHalf.h	Fri Nov 17 14:50:36 2017 -0800
@@ -231,7 +231,7 @@
 #else
   float a1 = __low2float(a);
   float a2 = __high2float(a);
-  return Eigen::half(half_impl::raw_uint16_to_half(__float2half_rn(a1 + a2)));
+  return Eigen::half(__float2half_rn(a1 + a2));
 #endif
 }
 
@@ -265,7 +265,7 @@
 #else
   float a1 = __low2float(a);
   float a2 = __high2float(a);
-  return Eigen::half(half_impl::raw_uint16_to_half(__float2half_rn(a1 * a2)));
+  return Eigen::half(__float2half_rn(a1 * a2));
 #endif
 }
 
