diff --git a/setup.py b/setup.py
index 9c73a10..e4107f1 100644
--- a/setup.py
+++ b/setup.py
@@ -253,7 +253,7 @@ def get_cuda_dirs(build_ext, cpp_flags):
     return cuda_include_dirs, cuda_lib_dirs
 
 
-def get_nccl_dirs(build_ext, cuda_include_dirs, cuda_lib_dirs, cpp_flags):
+def get_nccl_vals(build_ext, cuda_include_dirs, cuda_lib_dirs, cpp_flags):
     nccl_include_dirs = []
     nccl_lib_dirs = []
 
@@ -270,8 +270,14 @@ def get_nccl_dirs(build_ext, cuda_include_dirs, cuda_lib_dirs, cpp_flags):
     if nccl_lib:
         nccl_lib_dirs += [nccl_lib]
 
+    nccl_link_mode = os.environ.get('HOROVOD_NCCL_LINK') or ''
+    if nccl_link_mode.upper() == 'SHARED':
+        nccl_lib_name='nccl'
+    else:
+        nccl_lib_name='nccl_static'
+
     try:
-        test_compile(build_ext, 'test_nccl', libraries=['nccl_static'], include_dirs=nccl_include_dirs + cuda_include_dirs,
+        test_compile(build_ext, 'test_nccl', libraries=[nccl_lib_name], include_dirs=nccl_include_dirs + cuda_include_dirs,
                      library_dirs=nccl_lib_dirs + cuda_lib_dirs, extra_preargs=cpp_flags, code=textwrap.dedent('''\
             #include <nccl.h>
             #if NCCL_MAJOR < 2
@@ -292,7 +298,7 @@ def get_nccl_dirs(build_ext, cuda_include_dirs, cuda_lib_dirs, cpp_flags):
             'HOROVOD_NCCL_INCLUDE - path to NCCL include directory\n'
             'HOROVOD_NCCL_LIB - path to NCCL lib directory')
 
-    return nccl_include_dirs, nccl_lib_dirs
+    return nccl_include_dirs, nccl_lib_dirs, nccl_lib_name
 
 
 def get_common_options(build_ext):
@@ -323,7 +329,7 @@ def get_common_options(build_ext):
 
     if gpu_allreduce == 'NCCL':
         have_nccl = True
-        nccl_include_dirs, nccl_lib_dirs = get_nccl_dirs(
+        nccl_include_dirs, nccl_lib_dirs, nccl_lib_name = get_nccl_vals(
             build_ext, cuda_include_dirs, cuda_lib_dirs, cpp_flags)
     else:
         have_nccl = False
@@ -348,7 +354,7 @@ def get_common_options(build_ext):
         INCLUDES += nccl_include_dirs
         LINK_FLAGS += ['-Wl,--version-script=hide_nccl.lds']
         LIBRARY_DIRS += nccl_lib_dirs
-        LIBRARIES += ['nccl_static']
+        LIBRARIES += [nccl_lib_name]
 
     if gpu_allreduce:
         MACROS += [('HOROVOD_GPU_ALLREDUCE', "'%s'" % gpu_allreduce[0])]
